src/libopenrave-core/jsonparser/jsonreader.cpp

```diff
@@ -120,21 +112,17 @@ static void OpenRapidJsonDocument(const std::string& filename, rapidjson::Docume
     }
 }
 
-/// \brief get the scheme of the uri, e.g. file: or openrave:
-static void ParseURI(const std::string& uri, std::string& scheme, std::string& path, std::string& fragment)
+/// \brief open and cache an encrypted document
+static void OpenEncryptedDocument(const std::string& filename, rapidjson::Document& doc)
```

Comment:
I don't get this logic here. If we open gpg file, how do we know it's json inside? There is no context about json vs msgpack here.

Reply:
Yeah missed that. Added logic for using filename to detect.

Reply:
The point is: according to the design right now, I don't want to allow json parsing to switch to msgpack parsing in the middle of some reference chain. If top level is json, it has to be json all the way. If top leve is msgpack, it has to be msgpack all the way.